lib/facter/kmod.rb

```diff
@@ -0,0 +1,29 @@
+Facter.add(:kmods) do
+  confine :kernel => 'Linux'
+  confine File.exists?('/sys/module')
+
+  kmod = {}
+
+  setcode do
+    Dir.foreach('/sys/module') do |directory|
+      next if directory == '.' or directory == '..'
+
+      kmod[directory] = {}
+      kmod[directory]['parameters'] = {}
+      kmod[directory]['used_by'] = []
+
+      Dir.foreach("/sys/module/#{directory}/parameters") do |param|
+        next if param == '.' or param == '..'
+        kmod[directory]['parameters'][param] = File.open("/sys/module/#{directory}/parameters/#{param}", 'r').read
```

Comment:
Use `File.read("/sys/module/#{directory}/parameters/#{param}")` instead to avoid leaving a file handler open.